contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -269,6 +273,26 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     }
   }
 
+  /// @notice Removes nodes.  The node operator admin or contract owner
+  /// can remove nodes
+  /// @param removedNodeP2PIds The P2P Ids of the nodes to remove
+  function removeNodes(bytes32[] calldata removedNodeP2PIds) external {
+    bool isOwner = msg.sender == owner();
+    for (uint256 i; i < removedNodeP2PIds.length; ++i) {
+      bytes32 p2pId = removedNodeP2PIds[i];
+      Node memory node = s_nodes[p2pId];
+
+      bool nodeExists = s_nodes[p2pId].supportedHashedCapabilityIds.length > 0;
+      if (!nodeExists) revert InvalidNodeP2PId(p2pId);
+
+      NodeOperator memory nodeOperator = s_nodeOperators[node.nodeOperatorId];
+
+      if (!isOwner && msg.sender != nodeOperator.admin) revert AccessForbidden();
+      delete s_nodes[p2pId];
+      emit NodeRemoved(p2pId);
```

Comment:
Once DON management is complete, we must validate that the node doesn't belong to any DONs. It'd be great if you could create a ticket to keep track of it.